### PR TITLE
mpas-model: New package.

### PR DESCRIPTION
**Pull request by tkameyama**

### CODE REVIEW

File: var/spack/repos/builtin/packages/mpas-model/package.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    studies."""
+
+    homepage = "https://mpas-dev.github.io/"
+    url = "https://github.com/MPAS-Dev/MPAS-Model/archive/v7.0.tar.gz"
+
+    version('7.0', sha256='f898ce257e66cff9e29320458870570e55721d16cb000de7f2cc27de7fdef14f')
+    version('6.3', sha256='e7f1d9ebfeb6ada37d42a286aaedb2e69335cbc857049dc5c5544bb51e7a8db8')
+    version('6.2', sha256='2a81825a62a468bf5c56ef9d9677aa2eb88acf78d4f996cb49a7db98b94a6b16')
+
+    variant('pnetcdf', default=True, description='use parallel netcdf')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mpas-model/package.py]: It seems like this variant isn't necessary to change the build. Users can already specify this like so:
```console
$ spack install mpas-model ^parallelio+pnetcdf
```
I would remove this variant to keep the package simple.

File: var/spack/repos/builtin/packages/mpas-model/package.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    depends_on('mpi')
+    depends_on('parallelio +pnetcdf', when='+pnetcdf')
+    depends_on('parallelio ~pnetcdf', when='~pnetcdf')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mpas-model/package.py]:
```suggestion
    depends_on('parallelio')
```

File: var/spack/repos/builtin/packages/mpas-model/package.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        if satisfies('%gcc'):
+            fflags.extend([
+                '-ffree-line-length-none',
+                '-fconvert=big-endian',
+                '-ffree-form',
+                '-fdefault-real-8',
+                '-fdefault-double-8',
+            ])
+            cppflags.append('-DUNDERSCORE')
+        if satisfies('%fj'):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mpas-model/package.py]:
```suggestion
        elif satisfies('%fj'):
```

File: var/spack/repos/builtin/packages/mpas-model/package.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            'CC_SERIAL={0}'.format(spack_cc),
+            'CXX_SERIAL={0}'.format(spack_cxx),
+            'CFLAGS_OMP={0}'.format(self.compiler.openmp_flag),
+            'FFLAGS_OMP={0}'.format(' '.join(fflags)),
+            'CPPFLAGS={0}'.format(' '.join(cppflags)),
+            'PIO={0}'.format(spec['parallelio'].prefix),
+            'NETCDF={0}'.format(spec['netcdf-c'].prefix),
+            'NETCDFF={0}'.format(spec['netcdf-fortran'].prefix)
+        ]
+        if satisfies('+pnetcdf'):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mpas-model/package.py]:
```suggestion
        if satisfies('^parallelio+pnetcdf'):
```

File: var/spack/repos/builtin/packages/mpas-model/package.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            targets.append(
+                'PNETCDF={0}'.format(spec['parallel-netcdf'].prefix)
+            )
+        targets.extend([
+            'USE_PIO2=true', 'CPP_FLAGS=-D_MPI', 'OPENMP=true',
+            'CORE={0}'.format(model), action
+        ])
+        tty.debug(targets)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mpas-model/package.py]: Spack already logs the command that is run, I don't think this is needed.

File: var/spack/repos/builtin/packages/mpas-model/package.py (hunk 1 above)
Comment (tkameyama) [reply var/spack/repos/builtin/packages/mpas-model/package.py]: fixed.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**